src/rust-crypto/verification.ts

```diff
@@ -477,19 +483,39 @@ abstract class BaseRustVerifer<InnerType extends RustSdkCryptoJs.Qr | RustSdkCry
 
                 this.emit(VerificationRequestEvent.Change);
             };
-            inner.registerChangesCallback(onChange);
+            inner.registerChangesCallback(this.onChangeCallback);
         });
         // stop the runtime complaining if nobody catches a failure
         this.completionPromise.catch(() => null);
     }
 
+    /**
+     * Replace the inner Rust verifier with a different one.
+     *
+     * @param inner - the new Rust verifier
+     * @internal
+     */
+    public setInner(inner: InnerType): void {
```

Comment:
this should maybe be `replaceInner` rather than `setInner`, but that is somewhat gold-plating